Add function to indent things in blocks for nix-indent-line

etu opened the pull request:
Add a function `nix-indent-first-line-in-block` that checks if the last word on the previous line is one of "let, import, [, =, (, {". Then it adds two spaces of indent to the one of the previous line.

This is a shot at improving based on the examples is #60.

It works with all examples in #60 and just by going through my configuration.nix it seems a lot better than before.

It's not perfect in all the cases like this:
```nix
let
  splitString = _sep: _s: builtins.filter
    (x: builtins.typeOf x == "string")
    (builtins.split _sep _s);
in # whatever
```

But it wasn't before either, in fact it behaves exactly the same as before in that case as far as I can see.

grahamc commented:
What does it do in this case?

```
let
  splitString = _sep: _s: builtins.filter
    (x: builtins.typeOf x == "string")
    (builtins.split _sep _s);
in # whatever
```

etu commented:
@grahamc 
It becomes this:
```nix
let
  splitString = _sep: _s: builtins.filter
  (x: builtins.typeOf x == "string")
  (builtins.split _sep _s);
in # whatever
```

But before that would result in:
```nix
let
splitString = _sep: _s: builtins.filter
(x: builtins.typeOf x == "string")
(builtins.split _sep _s);
in # whatever
```

for me, so still better than before.

grahamc commented:
Confirming it fixes these cases:

```nix
{
  x86 = [
    {
      name = "t1.small.x86";
    }
  ];
}
```

```nix
let
  x = [
    (let y = 1; in y)
    { foo = 1; }
    [ 1 2 3 ]
    x
  ];
in x
```

(though I note on the first line, tabbing will indent `let` "forever")

```nix
let
  mozilla-overlay =
    import
  (
    builtins.fetchTarball
    https://github.com/mozilla/nixpkgs-mozilla/archive/master.tar.gz
  );
in mozilla-overlay
```

Looks like #60 can be closed with this merge!

matthewbauer commented:
Awesome! Thanks
